packages/fxa-settings/src/components/Settings/PageSettings/index.tsx

```diff
@@ -31,6 +34,18 @@ export const PageSettings = (_: RouteComponentProps) => {
     GleanMetrics.accountPref.view();
   }, []);
 
+  useEffect(() => {
+    // We want this view event to fire whenever the account settings page view
+    // event fires, if the user is shown the promo.
+    const { gleanEvent, hasAllProducts } = getProductPromoData(
+      attachedClients,
+      subscriptions
+    );
+    if (!hasAllProducts) {
+      GleanMetrics.accountPref.promoMonitorView(gleanEvent);
+    }
+  }, [attachedClients, subscriptions]);
```

Comment:
I was going to say I don't believe we ever update `attachedClients` or `subscriptions` within Settings so this should never fire more than we want it to, but of course we can remove devices so I think I need to keep track of that with a `useState`, will look now...